pkg/controller/kogitoapp/resource/deployment_config.go

```diff
@@ -194,54 +201,75 @@ func setReplicas(kogitoApp *v1alpha1.KogitoApp, dc *appsv1.DeploymentConfig) {
-			framework.SetEnvVarFromSecret(vars[envVarInfinispanPassword], infrastructure.InfinispanSecretPasswordKey, secret, &dc.Spec.Template.Spec.Containers[0])
+		appProps = map[string]string{}
+		envs = map[string]corev1.EnvVar{}
+
+		// inject credentials
+		vars := propertiesInfinispanQuarkus
+		if kogitoApp.Spec.Runtime == v1alpha1.SpringbootRuntimeType {
+			vars = propertiesInfinispanSpring
 		}
+
+		appProps[vars[appPropInfinispanUseAuth]] = "true"
+		appProps[vars[appPropInfinispanServerList]] = uri
+		appProps[vars[appPropInfinispanSaslMechanism]] = string(defaultInfinispanSaslMechanism)
+
+		envs[vars[envVarInfinispanUser]] = corev1.EnvVar{Name: vars[envVarInfinispanUser], ValueFrom: &corev1.EnvVarSource{
+			SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{Name: secret.Name},
+				Key:                  infrastructure.InfinispanSecretUsernameKey,
+			},
+		}}
+		envs[vars[envVarInfinispanPassword]] = corev1.EnvVar{Name: vars[envVarInfinispanPassword], ValueFrom: &corev1.EnvVarSource{
+			SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{Name: secret.Name},
+				Key:                  infrastructure.InfinispanSecretPasswordKey,
+			},
+		}}
 	}
-	return nil
+	return envs, appProps, nil
 }
 
-// SetKafkaEnvVars sets Kafka variables to the given KogitoApp instance DeploymentConfig by reading information from the KogitoInfra
-func SetKafkaEnvVars(cli *client.Client, kogitoInfra *v1alpha1.KogitoInfra, kogitoApp *v1alpha1.KogitoApp, dc *appsv1.DeploymentConfig) error {
-	if dc != nil && kogitoApp != nil &&
+// CreateKafkaProperties creates Kafka properties by reading information from the KogitoInfra
+func CreateKafkaProperties(cli *client.Client, kogitoInfra *v1alpha1.KogitoInfra, kogitoApp *v1alpha1.KogitoApp) (envs map[string]corev1.EnvVar, appProps map[string]string, err error) {
+	if kogitoApp != nil &&
 		(kogitoInfra != nil && &kogitoInfra.Status != nil && &kogitoInfra.Status.Kafka != nil) {
 		uri, err := infrastructure.GetKafkaServiceURI(cli, kogitoInfra)
 		if err != nil {
-			return err
+			return nil, nil, err
+		}
+
+		appProps = map[string]string{}
+		envs = map[string]corev1.EnvVar{}
+
+		vars := propertiesInfinispanQuarkus
+		if kogitoApp.Spec.Runtime == v1alpha1.SpringbootRuntimeType {
+			vars = propertiesInfinispanSpring
 		}
-		if len(dc.Spec.Template.Spec.Containers) > 0 {
-			framework.SetEnvVar(envVarKafkaBootstrapURI, uri, &dc.Spec.Template.Spec.Containers[0])
-			// let's also add a secret feature that injects all _BOOTSTRAP_SERVERS env vars with the correct uri :p
-			for _, env := range dc.Spec.Template.Spec.Containers[0].Env {
-				if strings.HasSuffix(env.Name, envVarKafkaBootstrapSuffix) {
-					framework.SetEnvVar(env.Name, uri, &dc.Spec.Template.Spec.Containers[0])
-				}
+
+		appProps[vars[appPropKafkaBootstrapServerList]] = uri
+		envs[envVarKafkaBootstrapURI] = corev1.EnvVar{Name: envVarKafkaBootstrapURI, Value: uri}
```

Comment:
Does it have sense to create an env variable with URI when it is supplied as application property?

Reply:
I am not sure if the job service works without this env according to the [scripts](https://github.com/kiegroup/kogito-images/blob/7a13116a98b7f6de9fba2ff61975106187ddc2c1/modules/kogito-jobs-service/added/launch/kogito-jobs-service.sh#L38), so I kept it.